D3D12Bundles/D3D12Bundles.cs

```diff
@@ -699,43 +651,41 @@ struct Vertex {
             // However, when ExecuteCommandList() is called on a particular command
             // list, that command list can then be reset at any time and must be before
             // re-recording.
-            mCommandList.Reset(mCurrentFrameResource.mCommandAllocator, mPipelineState1);
+            commandList.Reset(mCurrentFrameResource.mCommandAllocator, mPipelineState1);
 
             // Set necessary state.
-            mCommandList.SetGraphicsRootSignature(mRootSignature);
-
-            mCommandList.SetDescriptorHeaps(new[] { mCbvSrvHeap, mSamplerHeap });
+            commandList.SetNecessaryState(mRootSignature, mCbvSrvHeap, mSamplerHeap);
 
-            mCommandList.RSSetViewports(mViewport);
-            mCommandList.RSSetScissorRects(mScissorRect);
+            commandList.SetViewports(mViewport);
+            commandList.SetScissorRectangles(mScissorRect);
 
             // Indicate that the back buffer will be used as a render target.
-            mCommandList.ResourceBarrier(ResourceBarrier.BarrierTransition(mRenderTargets[mFrameIndex], ResourceStates.Present, ResourceStates.RenderTarget));
+            commandList.ResourceBarrierTransition(mPresenter.BackBuffer.Resource, ResourceStates.Present, ResourceStates.RenderTarget);
 
-            CpuDescriptorHandle rtvHandle = mRtvHeap.GetCPUDescriptorHandleForHeapStart().Offset(mFrameIndex, mRtvDescriptorSize);
-            CpuDescriptorHandle dsvHandle = mDsvHeap.GetCPUDescriptorHandleForHeapStart();
-            //mCommandList.OMSetRenderTargets(1, &rtvHandle, false, dsvHandle);
-            mCommandList.OMSetRenderTargets(rtvHandle, dsvHandle);
+            //Ultimately calls OMSetRenderTargets after updating CommandList.RenderTargets accordingly, however as not setting all render targets, can't use this since
+            //presenter doesn't expose all of them publicly.
+            //commandList.SetRenderTargets(commandList.DepthStencilBuffer, commandList.RenderTargets);
+            commandList.OMSetRenderTargets(mPresenter.BackBuffer.CpuDescriptorHandle, commandList.DepthStencilBuffer!.CpuDescriptorHandle);
```

Comment:
Here's another case, had to expose `OMSetRenderTargets` in order to do this without overriding CommandList.RenderTargets, which represents all 3 of them, even though we only want to set 1 of them here.

Because GraphicsPresenter which now owns them (and sets them on the command list upon construction) doesn't actually expose this, the use of `CommandList.SetRenderTargets` would irreversably remove the others from the collection, hence this code.  Hopefully I come up with an elegant solution to this mess in future.